server-spi/src/main/java/org/keycloak/models/UserProvider.java

```diff
@@ -40,6 +40,7 @@
 
     void addFederatedIdentity(RealmModel realm, UserModel user, FederatedIdentityModel socialLink);
     boolean removeFederatedIdentity(RealmModel realm, UserModel user, String socialProvider);
+    void removeFederatedIdentity(RealmModel realm, String socialProvider);
```

Comment:
Just minor concern, but from the name of this method, it is not obvious what is is doing... Is it possible to change the method signature to "preRemove(RealmModel, IdentityProviderModel)" to be compatible with all the other similar methods on the UserProvider interface (methods which intercepts removing on some realm object) ? WDYT?

Reply:
Sure

Reply:
@mposolda I've also added more assertion to tests.